### PR TITLE
Add API doc generation in CI (close #277)

### LINKED ISSUE #277: Add API doc generation in CI

### PR DESCRIPTION
**Pull request by Jack-Keene**
For issue 277

A Github Workflow that generates the API docs and adds them to the gh-pages branch has been added. The docs can be found here:

http://snowplow.github.io/snowplow-python-tracker/index.html

The docs folder shows in the main repo, I'm not sure if this is an issue?

### CODE REVIEW

File: docs/_build/html/_sources/index.rst.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+.. Snowplow Python Tracker documentation master file, created by
+   sphinx-quickstart on Wed Jul 20 14:00:53 2022.
+   You can adapt this file completely to your liking, but it should at least
+   contain the root `toctree` directive.
+
+Welcome to Snowplow Python Tracker's documentation!
```
Comment (matus-tomlein) [inline docs/_build/html/_sources/index.rst.txt:6]:
```suggestion
Snowplow Python Tracker API Documentation
```

Welcoming people to a website sounds so 90s 😅

File: docs/_build/html/_sources/index.rst.txt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+===================================================
+
```
Comment (matus-tomlein) [inline docs/_build/html/_sources/index.rst.txt:8]: Not that important but could we perhaps paste some short intro what is the tracker here? Maybe just copy the overview from the README, it looks a bit empty. Similar to [the front page of the C++ tracker API docs](https://snowplow.github.io/snowplow-cpp-tracker/).

**Review by matus-tomlein — CHANGES_REQUESTED** [review]
Looks great, really nice to have good API docs! 👍 

I added a few small comments for you to consider. Also, are you sure that the `docs/_build` folder needs to be added to Git? It seems like the output of the docs generation scripts. In that case, it would be better to remove it and make the repo a bit tidier.

There is one more thing that I have been thinking about – looking at other Spinx docs for Python libs (e.g., for Jinja, Flask), they often have some structure exposing the most useful classes up top for easier navigation. It's not that important as people can search within these docs, but do you think it could be possible to expose links to Tracker, Emitter, Subject, SelfDescribingJson on the front page? For example, [this is how the API page](https://flask.palletsprojects.com/en/2.1.x/api/) for Flask looks like and [here is their rst file](https://raw.githubusercontent.com/pallets/flask/main/docs/api.rst) that generates it. Don't know much about Sphinx, but do you think that we could pick some of the important classes up front like this:

```rst
.. autoclass:: Tracker
   :members:
   :inherited-members:
```

File: snowplow_tracker/celery/celery_emitter.py [hunk 3]
```diff
@@ -1,23 +1,23 @@
...
+#     Unless required by applicable law or agreed to in writing,
+#     software distributed under the Apache License Version 2.0 is distributed on
+#     an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+#     express or implied. See the Apache License Version 2.0 for the specific
+#     language governing permissions and limitations there under.
+
+#     Authors: Anuj More, Alex Dean, Fred Blundun, Paul Boocock
+#     Copyright: Copyright (c) 2013-2022 Snowplow Analytics Ltd
+#     License: Apache License Version 2.0
+# """
```
[the remark this answers is not in the document]
Comment (Jack-Keene) [reply snowplow_tracker/celery/celery_emitter.py:20]: Sphinx picks up anything with a string and assumes they're part of the docs, which made everything look messy. I'll go through and drop the strings!

**Review by matus-tomlein — APPROVED** [review]
LGTM!

**Pull request merged**